tests/plan/lint/discover.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+invalid_url() {
+    cat >invalid_url.fmf <<EOF
+discover:
+    how: fmf
+    url: http://invalid-url
+execute:
+    how: tmt
+EOF
+}
+
+invalid_ref() {
+    cat >invalid_ref.fmf <<EOF
+discover:
+    how: fmf
+    url: https://github.com/psss/tmt
+    ref: invalid-ref-123456
+execute:
+    how: tmt
+EOF
+}
+
+invalid_path() {
+    cat >invalid_path.fmf <<EOF
+discover:
+    how: fmf
+    url: https://github.com/psss/tmt
+    ref: master
+    path: /invalid-path-123456
+execute:
+    how: tmt
+EOF
+}
+
+invalid_how() {
+    cat >invalid_how.fmf <<EOF
+discover:
+    how: somehow
+execute:
+    how: tmt
+EOF
+}
+
+valid_fmf() {
+    cat >valid_fmf.fmf <<EOF
+discover:
+    how: fmf
+    url: https://github.com/psss/tmt
+    ref: master
+execute:
+    how: tmt
+EOF
+}
```

Comment:
Let's create these test data as a separate metadata tree under a [data](https://github.com/psss/tmt/tree/master/tests/prepare/adjust/data) subdirectory, similar as other tests do it. This allows easy experimenting in the directory when needed.

Reply:
on it

Reply:
will be resolved in next patchset